src/util/Simple8bCode.h

```diff
@@ -134,18 +134,24 @@ class Simple8bCode {
   // ! is necessary inside the decoding of a single codeword.
   template <typename Numeric>
   static void decode(uint64_t* encoded, size_t nofElements, Numeric* decoded) {
+    // Handle trivial empty case
+    if (!nofElements) {
+      return;
+    }
     uint64_t word;
     // Loop over full 64bit codewords and
-    for (size_t nofElementsDone(0), nofCodeWordsDone(0),
-         selector(encoded[nofCodeWordsDone] & SIMPLE8B_SELECTOR_MASK);
-         nofElementsDone < nofElements;
-         ++nofCodeWordsDone,
-         selector = encoded[nofCodeWordsDone] & SIMPLE8B_SELECTOR_MASK) {
+    size_t nofElementsDone(0), nofCodeWordsDone(0);
+    size_t selector(encoded[nofCodeWordsDone] & SIMPLE8B_SELECTOR_MASK);
+    while (nofElementsDone < nofElements) {
       word = encoded[nofCodeWordsDone] >> 4;
       for (size_t i(0); i < SIMPLE8B_SELECTORS[selector]._groupSize; ++i) {
         decoded[nofElementsDone++] = word & SIMPLE8B_SELECTORS[selector]._mask;
         word >>= SIMPLE8B_SELECTORS[selector]._itemWidth;
       }
+      ++nofCodeWordsDone;
+      if (nofElementsDone < nofElements) {
+        selector = encoded[nofCodeWordsDone] & SIMPLE8B_SELECTOR_MASK;
+      }
```

Comment:
Nice refactoring!
Just to my understanding:
- The loops are almost equivalent, yours is just better readable.
- The original code had an off-by-one error (`selector = encoded[nofCodeWordsDone] & SIMPLE8B_SELECTOR_MASK;` was also executed one last time but never used.

Reply:
Yes, that's how I understand it as well